PSICompile/ILCodeGen.cs

```diff
@@ -66,9 +93,37 @@ public class ILCodeGen : Visitor {
       }
       if (w.NewLine) Out ("    call void [System.Console]System.Console::WriteLine ()");
    }
-   
-   public override void Visit (NIfStmt f) => throw new NotImplementedException ();
-   public override void Visit (NForStmt f) => throw new NotImplementedException ();
+
+   public override void Visit (NIfStmt f) {
+      string fLabel = NextLabel (), tLabel = NextLabel ();
+      f.Condition.Accept (this);
+      Out ($"    brfalse {fLabel}");
+      f.IfPart.Accept (this);
+      Out ($"    br {tLabel}");
+      Out ($"  {fLabel}:");
+      f.ElsePart?.Accept (this);
+      Out ($"  {tLabel}:");
+   }
+   public override void Visit (NForStmt f) {
+      f.Start.Accept (this);
+      var vd = (NVarDecl)mSymbols.Find (f.Var)!;
+      var type = TMap[vd.Type];
+      StoreVar (f.Var);
+      var cLabel = NextLabel ();
+      var bLabel = NextLabel ();
+      Out ($"    br {cLabel}");
+      Out ($"    {bLabel}:");
+      f.Body.Accept (this);
+      Out ($"    {(vd.Local ? "ldloc " : $"ldsfld {type} Program::")}{vd.Name}");

```

Comment:
You could write LoadVar similar to StoreVar

Reply:
Added LoadVar helper method